Adding German translation

deining opened the pull request:
This PR brings the German translation of the GUI and error texts to the project.

TheBizzle commented:
I speak German, but surely not as well as you.  Since it seems that your contributions are most likely generated by LLMs, I'd like to question some of the translations given here, despite the fact that you likely know the German language far better than I do.

The translation in this PR seems mostly suitable, but I don't understand the use of "Akteur" for "agent" (a very important word in the realm of NetLogo).  The German word "Agent" seems like an obvious choice here.  I have read [a German-language article about NetLogo published in 2003](https://klaus-manhart.de/.cm4all/uproc.php/0/netlogo-ct.pdf?cdp=a&_=188b6182a28), where the author uses "Agent" and "Agenten" dozens of times, but never "Akteur".

Similarly, I'm not entirely sure about your translations of "turtle" to "Schildkröte", "patch" to "Patch", and "link" to "Verbindung".  I get that it can be tricky to decide when to translate something and when to use Denglisch.  I think it's notable that the author of the article I mentioned seems to switch between calling patches "Patches" and "Zellen", and maybe either could be considered correct.  But, when it comes to turtles (another very fundamental word in NetLogo), the author uses "Schildkröte" 8 times, vs. 89 usages of "Turtle" (about ten of which are in code and should be discounted).  Links go unmentioned in the article (as I think it was published a few years before links were implemented in NetLogo).

My intuition is that "Akteur", "Schildkröte", and "Verbindung" are not the proper German terms for us to be using to refer to these concepts, and that "Agent", "Turtle", and "Link" should be used instead.  But, as the person who is presumably a native German speaker, I'd be interested to hear your thoughts.

deining commented:
Hi Jason,

thanks for your comments and your valuable feedback, I appreciate you start a discussion here to end up with the best possible translation. Through your comments, it became clear to me that we may not fully there yet. But let my start with a few facts.

> I speak German, but surely not as well as you.

Yes, I'm a native speaker and I do have quite some knowledge in the field of programming a associated concepts.

> Since it seems that your contributions are most likely generated by LLMs

No. I only used LLM for pre-translation. I had a look at each single translation phrase. To get the context, I tried to identify the place in the program where the translation occurs, and then I adapted the pretranslation to the context. And believe me, I made a lot of adjustments to get a good, natural translation. Especially translating the error texts was all but easy!

> My intuition is that "Akteur", "Schildkröte", and "Verbindung" are not the proper German terms for us to be using to refer to these concepts, and that "Agent", "Turtle", and "Link" should be used instead.

You a adressing a very valid and central point, and believe me, I thought about this too. And yes, I also thought about using `Agent`, `Turtle`, and `Link`. But I deliberately decided to **not** use these terms, mainly for two reasons:

- **audience/target group**: I wanted to do some fun programming with my daughter, 10 years old. This is how I came in contact with NetLogo. When I started to use NetLogo together with her, I immediately realized that she had difficulties when using the English NetLogo GUI interface. That's what motivated me to start with the translation. And very soon, I came to the point where I had to translate terms like `link`, ... . Pretty soon I realized it wouldn't help my daughter if we leave these terms as is (using Denglish). That's basically the reason why I'm using German terms here. Now you can argue that NetLogo does not address kids in the first plae, but is meant as simulation tool, even for scientific purposes. I we have this audience in mind, it woul be certainly better to use the English terms. But members of these target group can switch to English anyway, without loosing too much comfort. But this is different for kids: if we use German phrases for the keywords, they will feel much more comfortable and will be able to gras conecpts much easier. And switching back to the original English translation is **no** choice for kids, obviously.
- another reason is that unfortunately, almost all of these central terms convey a **different meaning** if we translate them 1:1 to German (even to me as a person that is quite savy with programming). Let me go through the terms inividually:
    - `Agent`: colloquially if you talk about an `Agent`, the first meaning that will come to the mind of a German speaker (and especially kids) will be `spy` (James Bond 007 ist ein Agent!). And yes, there is a second meaning of `Agent` in the sense of `Agentenbasierte Modellierung`. But I think the term `Akteur` conveys the meaning much better here.
    - `Turtle`: this is another story, The correct translation `Schildkröte` is quite a long word in German, and I've seen German Logo variants adressing kids that used the term `Igel` (hedgehog) instead of `Schildkröte`, just to make typing easier! The problem here is that `Turtle` is not a German word at all (you won't find it in [Duden](https://www.duden.de)!). And chances that especially kids don't have any glue what a `Turtle` is are pretty high.
    - `Patch`: that's a difficult term, too. I left it as patch, but only because I could't find a better German term. The article you are referring to talks about `Zelle(n)` which is much better. If we decide to leave the German translations in place, I would like to change `patch` to `Zelle`, this would make my approach even more consistent.
    - `Link`: In German the term `Link` exists, but if you use it, you are mostly talking about a hyperlink (`<a href="..">`), at least that's what I associate with the term `Link`. In that sense, I was unhappy with using the term `Link` as is and deliberately chose `Verbindung` here, which is the better term here (if we disregard the fact that this term is different from the English term).
    - `Ticks`: that's another difficult phrase. There is a verb `ticken` in German, but the noun is `Ticken` (das `Ticken` der Uhr). A `Tick` mostly refers to some abnormal behaviour of a person and is **not** really associated with ticks in the sense of time related, synchronous action.  

This basically explains how I came up with my translation an why it is as it is. To make a long story short, there are/were good reasons for all my choices. But I see your points, too. Bascially we have to make a decision: if we have kids as audience in mind, my translation seems much better to me. If we are targeting at adults, your proposed changes make a lot of sense. I tend to leave my translation as-is (except for replacing `patches` with `Zellen`). But if you want we to use the English phrases, I will change my PR according to your wished and can live with this easily.

WDYT? I'm doing quite some translation stuff as part of my profession, and in my experinence you sometimes have to make a choice because there are good reasons for more than one translation variant.

deining commented:
> Your points make a lot of sense.

Glad to hear that.

> So I'm convinced by your point that it's more sensible to provide a translation that is intuitive for children. We can keep the translations of "Schildkröte" and "Verbindung". And feel free to make the "Zelle" change.

The "Zelle" change is part of the PR meanwhile.

 > The one I'm less sure about is the translation of "agent" as "Akteur". I accept that "Akteur" is more intuitive, and avoids evoking the wrong ideas. But NetLogo is the most widely-used Agentenbasierte Modellierung software. And that's a major part of NetLogo's identity.

I see your point. I changed all occurrences of `Akteure` to `Agenten` with one single exception, where I gave both phrases:

```
edit.button.agents = Agent(en)/Akteur(e)
```

I hope this is fine for you.

> Regarding "ticks", I feel like I would understand "Zeitschritten" as "ticks" if I came across it, but maybe that's not an idiomatic translation.

Thanks for your suggestion, very helpful. I'm now translating `ticks` as `Zeittakte`, and I'm quite happy with this translation.
Furthermore I changed the translation of `observer` from `Beobachter` to `Betrachter`.

Thanks for your suggestions which brought in some new ideas that improved the overall quality of this PR. I consider this PR ready for merging now. Or, please let me know if there are further changes needed.

deining commented:
> Very cool 😎 , thank you for your contribution! 🙇‍♂️ 

Glad I could help out here. Just tried Netlogo 7 (preview) with my daughter yesterday, and it turned out it makes a big difference to have it in German now 😄. There may be some edges, but overall, I think the translation is good.

That brings me to another point: I tried [Turtle Universe](https://turtlesim.com/products/turtle-universe/) recently, just to realize that their German translation is crap and unuseable.

Example`from the start screen:

![grafik](https://github.com/user-attachments/assets/c93136dc-0d13-4c68-af7c-f2fd25535970)

Translating `Workspace` with `Bergwerk` is ridiculous / funny. I would recommend to to disable the option to select `German` from the app settings. In the current form, it's unusable. I immediately swicthed back to English, this was hard to stand!

I don't know anything about the development status of TurtleUniverse, I'm not even sure if you are affiliated with TurtleUniverse at all (and if yes in what way). I'm not really keen on translating another NetLogo variant, but if you interested in getting a solid German translation for TurtleUniverse, please let me know.

>I always wanted to give NetLogo German language support, but didn't want people to end up with translations like "Zeitschritte". 😬

Actually `Zeitschritte` is not that bad, but `Zeittakte` is probably the better translation term here. 😀 

Thanks for considering my PR, your comments helped me a lot to come up with a really good result IMHO.

TheBizzle commented:
> Translating `Workspace` with `Bergwerk` is ridiculous / funny.

That's actually a hilarious (non-)translation, I love it! 🤣

> I'm not even sure if you are affiliated with TurtleUniverse at all (and if yes in what way)

I have no direct involvement in Turtle Universe.  It is primarily the work of @CIVITAS-John, who is a graduate student in our lab.  I think you can open a ticket with him [here](https://github.com/NetLogo-Mobile/Turtle-Universe-Issue-Tracker/issues) if you're interested in bringing the issue to his attention.

CIVITAS-John commented:
@deining I would love to have our German translation fixed. Please let me know and I can bring you to the source code repo (not publicly available as yet, but work is being done to make it happen)

deining commented:
> @deining I would love to have our German translation fixed.

Great. As stated above, I'm willing to contribute.

> Please let me know and I can bring you to the source code repo

Is the repo a private repo on GitHub? If so, please send me an invitation and I will accept it. If the repo is not hosted on GitHub, please let me know how to get access to the sources.
